src/stepFunctions/commands/visualizeStateMachine/aslVisualizationCDKManager.ts

```diff
@@ -0,0 +1,108 @@
+
+        const cdkOutPath = node.id.replace(`/tree.json/${node.tooltip}`, ``)
+        const workspaceName = this.getCDKAppWorkspaceName(cdkOutPath)
+
+        const existingVisualization = this.getExistingVisualization(workspaceName, node.tooltip)
+
+        if (existingVisualization) {
+            existingVisualization.showPanel()
+
+            return existingVisualization.getPanel()
+        }
+
+        const stateMachineName = node.label
+        const appName = node.tooltip.replace(`/${stateMachineName}`, ``)
+        const templatePath = normalize(`${cdkOutPath}/${appName}.template.json`)
+        const uri = vscode.Uri.file(templatePath)
+
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await this.cache.updateCache(globalStorage)
+
+            const textDocument = await vscode.workspace.openTextDocument(uri)
+            const newVisualization = new AslVisualizationCDK(textDocument, templatePath, appName, stateMachineName)
+            if (newVisualization) {
+                this.handleNewVisualization(workspaceName, newVisualization)
+                return newVisualization.getPanel()
+            }
+        } catch (err) {
+            this.handleErr(err as Error, logger)
+        }
+
+        return
+    }
+
+    protected handleNewVisualization(workspaceName: string, newVisualization: AslVisualizationCDK): void {
+        let map = this.managedVisualizations.get(workspaceName)
+        const uniqueIdentifier = newVisualization.cdkAppName + '/' + newVisualization.stateMachineName
+        if (!map) {
+            map = new Map<string, AslVisualizationCDK>()
+            map.set(uniqueIdentifier, newVisualization)
+            this.managedVisualizations.set(workspaceName, map)
+        } else {
+            map.set(uniqueIdentifier, newVisualization)
+        }
+
+        const visualizationDisposable = newVisualization.onVisualizationDisposeEvent(() => {
+            this.deleteVisualization(workspaceName, uniqueIdentifier)
+        })
+        this.pushToExtensionContextSubscriptions(visualizationDisposable)
+    }
+
+    public getManagedVisualizations(): Map<string, Map<string, AslVisualizationCDK>> {
+        return this.managedVisualizations
+    }
+
+    protected deleteVisualization(workspaceName: string, visualizationToDelete: string): void {
+        this.managedVisualizations.get(workspaceName)?.delete(visualizationToDelete)
+    }
+
+    protected getExistingVisualization(
+        workspaceName: string,
+        uniqueIdentifier: string
+    ): AslVisualizationCDK | undefined {
+        return this.managedVisualizations.get(workspaceName)?.get(uniqueIdentifier)
+    }
+
+    /**
+     * @param {string} cdkOutPath - path to the cdk.out folder
+     * @returns name of the CDK application workspace name
+     */
+    public getCDKAppWorkspaceName(cdkOutPath: string): string {
+        if (typeof cdkOutPath !== 'string') {
+            return cdkOutPath
+        }
```

Comment:
Under what cases would this get hit, and if you're hitting a non-string, why would you return it?

Reply:
I wonder if we can have TypeScript find dead-code like this for us. TypeScript would mark `cdkOutPath` as having the `never` type here since it's not possible.

Reply:
Don't think this does anything since the parameter is typed anyways. Removed it.